disttask/framework/planner/plan.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package planner
+
+import (
+	"context"
+
+	"github.com/pingcap/tidb/sessionctx"
+)
+
+// PlanCtx is the context for planning.
+type PlanCtx struct {
+	Ctx context.Context
+
+	// integrate with current distribute framework
+	SessionCtx sessionctx.Context
+	TaskKey    string
+	TaskType   string
+	ThreadCnt  int
+
+	// PreviousSubtaskMetas is a list of subtask metas from previous step.
+	// We can remove this field if we find a better way to pass the result between steps.
+	PreviousSubtaskMetas [][]byte
```

Comment:
hold task_table as member?

Reply:
Using PreviousSubtaskMetas is to express that PreviousSubtaskMetas is the result of the previous stage. If we use task_table, our planner will be coupled with the system table again, although adding a new interface would be more convenient. 🤔